karamel-core/src/main/java/se/kth/karamel/backend/machines/SshMachine.java

```diff
@@ -183,8 +183,12 @@ private void runTask(Task task) {
           } else {
             try {
               task.collectResults(this);
+              // If this task is an experiment, try and download the experiment results
+              if (cmd.getCmdStr().contains("experiment") && cmd.getCmdStr().contains("json")) {
```

Comment:
I think this condition might be true for some non-experiment tasks as well, how about calling the "downloadExperimentResults" inside collect result of ExperimentTask?